docs/contributing/backend.md

```diff
@@ -0,0 +1,183 @@
+
+To understand how the backend fits into the broader system, we recommend reviewing the following resources:
+
+* [General Concepts](../admin/infrastructure/validated-architectures/index.md#general-concepts): Essential concepts and language used to describe how Coder is structured and operated.
+
+* [Architecture](../admin/infrastructure/architecture.md): A high-level overview of the infrastructure layout, key services, and how components interact.
+
+These sections provide the necessary context for navigating and contributing to the backend effectively.
+
+## Tech Stack
+
+Coder's backend is built using a collection of robust, modern Go libraries and internal packages. Familiarity with these technologies will help you navigate the codebase and contribute effectively.
+
+### Core Libraries & Frameworks
+
+* [go-chi/chi](https://github.com/go-chi/chi): lightweight HTTP router for building RESTful APIs in Go
+* [golang-migrate/migrate](https://github.com/golang-migrate/migrate): manages database schema migrations across environments
+* [coder/terraform-config-inspect](https://github.com/coder/terraform-config-inspect) *(forked)*: used for parsing and analyzing Terraform configurations, forked to include [PR #74](https://github.com/hashicorp/terraform-config-inspect/pull/74)
+* [coder/pq](https://github.com/coder/pq) *(forked)*: PostgreSQL driver forked to support rotating authentication tokens via `driver.Connector`
+* [coder/tailscale](https://github.com/coder/tailscale) *(forked)*: enables secure, peer-to-peer connectivity, forked to apply internal patches pending upstreaming
+* [coder/wireguard-go](https://github.com/coder/wireguard-go) *(forked)*: WireGuard networking implementation, forked to fix a data race and adopt the latest gVisor changes
+* [coder/ssh](https://github.com/coder/ssh) *(forked)*: customized SSH server based on `gliderlabs/ssh`, forked to include Tailscale-specific patches and avoid complex subpath dependencies
+* [coder/bubbletea](https://github.com/coder/bubbletea) *(forked)*: terminal UI framework for CLI apps, forked to remove an `init()` function that interfered with web terminal output
+
+### Coder libraries
+
+* [coder/terraform-provider-coder](https://github.com/coder/terraform-provider-coder): official Terraform provider for managing Coder resources via infrastructure-as-code
+* [coder/websocket](https://github.com/coder/websocket): minimal WebSocket library for real-time communication
+* [coder/serpent](https://github.com/coder/serpent): CLI framework built on `cobra`, used for large, complex CLIs
+* [coder/guts](https://github.com/coder/guts): generates TypeScript types from Go for shared type definitions
+* [coder/wgtunnel](https://github.com/coder/wgtunnel): WireGuard tunnel server for secure backend networking
+
+## Repository Structure
+
+The Coder backend is organized into multiple packages and directories, each with a specific purpose. Here's a high-level overview of the most important ones:
```

Comment:
Great detail! Wdyt about adding a key areas of interest section under here as well, just so people scanning this can get a better idea of what's the _most_ important. Maybe point out agent, coderd, queries, provisioner for instance.

Reply:
> Maybe point out agent, coderd, queries, provisioner for instance.

I'm intrigued! Do you have something specific in mind? Should I move up to most important ones?